src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq_vmr.c

```diff
@@ -3385,31 +3389,28 @@ static int xgq_vmr_remove(struct platform_device *pdev)
 	return 0;
 }
 
-static bool vmr_check_sc_is_ready(struct xocl_xgq_vmr *xgq)
-{
-	struct xgq_cmd_cq_vmr_payload *vmr_status =
-		(struct xgq_cmd_cq_vmr_payload *)&xgq->xgq_cq_payload;
-	int ret = vmr_status_query(xgq->xgq_pdev);
-
-	if (ret)
-		XGQ_ERR(xgq, "received error %d for vmr_status_query xgq request", ret);
-
-	if (vmr_status->sc_is_ready)
-		return true;
-
-	return false;
-}
-
 /* Wait for SC is fully ready during driver init (in reset) */
 static bool vmr_wait_for_sc_ready(struct xocl_xgq_vmr *xgq)
 {
 	const unsigned int loop_counter = vmr_sc_ready_timeout *
 		(1000 / SC_WAIT_INTERVAL_MSEC);
 	unsigned int i = 0;
+	struct xgq_cmd_cq_vmr_payload *vmr_status = (struct xgq_cmd_cq_vmr_payload *)&xgq->xgq_cq_payload;
+	int ret = 0;
+
+	vmr_status_query(xgq->xgq_pdev);
+	if (ret)
+		XGQ_ERR(xgq, "received error %d for vmr_status_query xgq request", ret);
+
+	if (!vmr_status->has_ext_scfw)
```

Comment:
You can enhance the vmr_check_sc_is_ready to 2 apis
vmr_check_sc_is_available {}
vmr_check_sc_is_ready {
   if (vmr_check_sc_is_available)
      return -ENDEV;

   return vmr_status->sc_is_ready;
}
in your previous code, you can just call vmr_check_is_is_avaialbe
and call vmr_check_sc_is_ready -> vmr_check_sc_is_available here

Reply:
vmr_check_sc_is_ready () is being called by vmr_wait_for_sc_ready() which is where the retires are being performed and this retries need to be skipped when SC is not available. calling vmr_check_sc_is_available will not help skipping the retry mechanism so moved it out of vmr_check_sc_is_ready()